data_model/src/events/data/events.rs

```diff
@@ -412,11 +445,32 @@ mod trigger {
             match self {
                 Self::Created(id)
                 | Self::Deleted(id)
-                | Self::Extended(id)
-                | Self::Shortened(id) => id,
+                | Self::Extended(TriggerNumberOfExecutionsChangedBy { trigger_id: id, .. })
+                | Self::Shortened(TriggerNumberOfExecutionsChangedBy { trigger_id: id, .. }) => id,
             }
         }
     }
+
+    /// [`TriggerNumberOfExecutionsChangedBy`] depending on wrapping event represents increased/decreased number of event executions
+    #[derive(
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Debug,
+        Hash,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[allow(missing_docs)]
+    pub struct TriggerNumberOfExecutionsChangedBy {
+        pub trigger_id: TriggerId,
+        pub by: u32,
```

Comment:
Either rename the `by` or remove the `By` suffix. I know what you wanted to do, but `trigger_number_of_executions_changed_by.by` isn't pretty.